### ISSUE TITLE
* is omitted from result set when query has other fields

### ISSUE THREAD

**Issue by galkk** [opendistro-for-elasticsearch/sql#101]
When columns other than * are present, the star is ommitted

```
POST _opendistro/_sql
{
  "query": "SELECT *,utc_time as t from kibana_sample_data_logs limit 1"
}
```

Expected results:
All fields are present, with utc_time having alias

Actual result:
Only utc_time is present

**Comment by galkk** [comment]
Obviously, current behavior is most confusing, but, I recommend to investigate and define expected behavior. 
As of now there seem to be at least 2 ways:

1. Traditional relational databases approach, that describes the expected behavior that I've represented here
http://sqlfiddle.com/#!17/4f55f/1

2. Elastic and PartiQL actually throw an exception for such query

###Elastic:
```
{
  "error": {
    "root_cause": [
      {
        "type": "sql_illegal_argument_exception",
        "reason": "Arrays (returned by [tags]) are not supported"
      }
    ],
    "type": "sql_illegal_argument_exception",
    "reason": "Arrays (returned by [tags]) are not supported"
  },
  "status": 500
```

### PartiQL:
```
Other expressions may not be present in the select list when '*' is used without dot notation.
```

**Comment by dai-chen** [comment]
It's doable to return alias for JDBC format because we can do the post-processing during conversion from ES response to our own class. For default format, if I understand this right, ES can only return all fields at most, which means we need to modify ES response DSL to add one or more alias. If so, I propose we return correct alias for JDBC format but throw exception for default format.
